dateparser/freshness_date_parser.py

```diff
@@ -5,235 +5,51 @@
-            }
-        },
-        'cs': {
-            'units': {
-                'year':     ('rok', 'roků'),
-                'month':    ('měsíc', 'měsíců', 'měsíce'),
-                'week':     ('týden', 'týdnů'),
-                'day':      ('den', 'dnů', 'dny'),
-                'hour':     ('hodina', 'hodin', 'hodiny', 'hodinami'),
-                'minute':   ('minuta', 'minut'),
-            }
-        },
-        'cn': {
-            'word_replacements': [
-                ('1天', ['昨天']),
-                ('2天', ['前天']),
-            ],
-            'units': {
-                'year':     ('年',),
-                'month':    ('月', '个月'),
-                'week':     ('周', '星期'),
-                'day':      ('天',),
-                'hour':     ('小时',),
-                'minute':   ('分', '分钟'),
-            },
-            'no_word_spacing': True,
-        },
-        'ar': {
-            'word_replacements': [
-                ('0 يوم', ['اليوم']),
-                ('1 يوم',['يوم أمس']),
-                ('2 يوم', ['يومين']),
-                (r'\1 أيام', [r'\b(\d+)\s*أيام']),
-            ],
-            'units': {
-                'year':     ('عام', 'سنة'),
-                'month':    ('شهر',),
-                'week':     ('أسبوع',),
-                'day':      ('يوم', 'أيام'),
-                'hour':     ('ساعة', 'ساعات'),
-                'minute':   ('دقيقة', 'دقائق'),
-            },
-        },
-        'th': {
-            'word_replacements': [
-                (u'0 วัน', [u'วันนี้']),
-                (u'1 วัน', [u'เมื่อวานนี้', u'1 วันที่แล้ว']),
-                (u'2 วัน', [u'เมื่อวานซืน', u'2 วันที่แล้ว'])
-            ],
-            'units': {
-                'year':     (u'ปี',),
-                'month':    (u'เดือน',),
-                'week':     (u'สัปดาห์',),
-                'day':      (u'วัน',),
-                'hour':     (u'ชั่วโมง', u'ชม.'),
-                'minute':   (u'นาที',),
-                'second':   (u'วินาที',),
-            },
-            'no_word_spacing': True,
-        }
-    }
+    """ Parses date string like "1 year, 2 months ago" and "3 hours, 50 minutes ago" """
 
     def __init__(self, now=None):
         self.now = now or datetime.utcnow()
-        self.units_map = {}
 
-        for lang in self.langs.itervalues():
-            d = lang['units']
-            for k, vlist in d.iteritems():
-                for v in vlist:
-                    self.units_map[v.lower()] = k
+    def _are_all_words_units(self, date_string):
+        skip = [_UNITS,
+                r'about|ago|\d+',
+                r':|[ap]m'] 
+
+        date_string = re.sub(r'\s+', ' ', date_string.strip())
+
+        words = filter(lambda x: x if x else False, re.split('\W', date_string))
+        words = filter(lambda x: not re.match(r'%s' % '|'.join(skip), x), words)
+        return not bool(words)
+
+    def _parse_time(self, date_string):
```

Comment:
Could we have a docstring here explaining why this is needed?
If I understood correctly, this is for dates like "yesterday 2pm", with yesterday replaced by "2 days", right?

Something like:

    """Attempts to parse the time part for date strings like '1 day ago at 2pm'"""